### PR TITLE
RAVTDBQ-123 Support for unpivot

### PR DESCRIPTION
**Pull request by Abhi-210793**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rel/rel2sql/UnpivotRelToSqlUtil.java [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+   *                     Arrays.asList(ImmutableList.of(builder.literal("jan")),//column_alias
+   *              ImmutableList.of(builder.literal("feb")),
+   *              ImmutableList.of(builder.literal("march"))),
+   *                     Arrays.asList(ImmutableList.of(builder.field("jan_sales")),//column_list
+   *              ImmutableList.of(builder.field("feb_sales")),
+   *              ImmutableList.of(builder.field("mar_sales")))))
+   *      .build();
+   * }</pre></blockquote>
+   *
+   * <p> Rel with includeNulls = true after expansion
```
Comment (pranav-joshi-dm) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/UnpivotRelToSqlUtil.java:191]: Ending tag for \<p> missed, plz check.

File: core/src/main/java/org/apache/calcite/rel/rel2sql/UnpivotRelToSqlUtil.java [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+  }
+
+  /**
+   * Check if case pattern & logical values pattern are equivalent to unpivot expansion.
+   * ex- case when month='jan' then jan_sales
+   * when month='feb' then feb_sales
+   * when month='mar'  then march_sales
+   * else null
+   * AS monthly_sales
+   * <p>
```
Comment (pranav-joshi-dm) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/UnpivotRelToSqlUtil.java:267]: Please check if it's a typo '\<p>'?

File: core/src/main/java/org/apache/calcite/rel/rel2sql/UnpivotRelToSqlUtil.java [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+ * Class to identify Rel structure which is of Unpivot type.
+ */
+public class UnpivotRelToSqlUtil {
+  Map<String, SqlNodeList> caseAliasVsThenList = new HashMap<>();
+
+  /**
+   * <p> SQL. </p>
+   * <blockquote><pre>{@code
+   *  SELECT *
+   *        FROM emp
```
[the remark this answers is not in the document]
Comment (Abhi-210793) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/UnpivotRelToSqlUtil.java:61]: done

**Comment by pranav-joshi-dm** [comment]
Looks good. L1 review completed.

**Review by deepakkadam-dm — APPROVED** [review]
Looks ok

**Pull request merged**